pkg/internal/nodeinfo/cpu_models_test.go

```diff
@@ -0,0 +1,164 @@
+							corev1.ResourceMemory: resource.MustParse("64Gi"),
+						},
+						Allocatable: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("15"),
+							corev1.ResourceMemory: resource.MustParse("60Gi"),
+						},
+					},
+				},
+				&corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "medium-node",
+						Labels: map[string]string{
+							nodeinfo.LabelNodeRoleWorker:           "",
+							"cpu-model.node.kubevirt.io/Haswell":   "true", // Medium PassMark (7200) - appears on 2 nodes
+							"cpu-model.node.kubevirt.io/Broadwell": "true", // Medium PassMark (6800)
+						},
+					},
+					Status: corev1.NodeStatus{
+						NodeInfo: corev1.NodeSystemInfo{
+							Architecture: "amd64",
+						},
+						Capacity: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("8"),
+							corev1.ResourceMemory: resource.MustParse("32Gi"),
+						},
+						Allocatable: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("7"),
+							corev1.ResourceMemory: resource.MustParse("30Gi"),
+						},
+					},
+				},
+				&corev1.Node{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "weak-node",
+						Labels: map[string]string{
+							nodeinfo.LabelNodeRoleWorker:        "",
+							"cpu-model.node.kubevirt.io/Penryn": "true", // Low PassMark (2400), only on weak node
+						},
+					},
+					Status: corev1.NodeStatus{
+						NodeInfo: corev1.NodeSystemInfo{
+							Architecture: "amd64",
+						},
+						Capacity: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("2"),
+							corev1.ResourceMemory: resource.MustParse("4Gi"),
+						},
+						Allocatable: corev1.ResourceList{
+							corev1.ResourceCPU:    resource.MustParse("1.5"),
+							corev1.ResourceMemory: resource.MustParse("3Gi"),
+						},
+					},
+				},
+			}
+
+			ctx := context.Background()
+			cl := fake.NewClientBuilder().WithScheme(scheme).WithObjects(nodes...).Build()
+			_, err := nodeinfo.HandleNodeChanges(ctx, cl, nil, GinkgoLogr)
+			Expect(err).ToNot(HaveOccurred())
+
+			recommendedCpuModels = nodeinfo.GetRecommendedCpuModels()
+			Expect(recommendedCpuModels).ToNot(BeEmpty())
+
+			modelMap = make(map[string]v1beta1.CpuModelInfo)
+			for _, model := range recommendedCpuModels {
+				modelMap[model.Name] = model
+			}
+		})
+
+		It("should exclude models that appear only once on a weak node", func() {
+			modelNames := make([]string, len(recommendedCpuModels))
+			for i, model := range recommendedCpuModels {
+				modelNames[i] = model.Name
+			}
+			Expect(modelNames).ToNot(ContainElement("Penryn"))
+		})
+
+		It("should correctly aggregate CPU across nodes", func() {
+			if skylake, exists := modelMap["Skylake"]; exists {
+				Expect(skylake.Cpu).To(Equal("16"))
+			}
+			if haswell, exists := modelMap["Haswell"]; exists {
+				Expect(haswell.Cpu).To(Equal("24")) // 16 + 8
+			}
+			if broadwell, exists := modelMap["Broadwell"]; exists {
+				Expect(broadwell.Cpu).To(Equal("8"))
+			}
```

Comment:
Since we control the input, I think we should expect the map keys, because if we did something wrong, and we won't have these key in the map, the test won't find them.

WDYT about this (two options)?

```suggestion
			Expext(odelMap).To(HaveKey("Skylake")
			Expect(modelMap["Skylake"].Cpu).To(Equal("16"))
			// OR
			haswell, exists := modelMap["Haswell"]
			Expect(exists).To(BeTrue())			
			Expect(haswell.Cpu).To(Equal("24")) // 16 + 8
			...
```